### ISSUE TITLE
Bug in EuroSAT.plot()

### ISSUE THREAD

**Issue by robmarkcole** [microsoft/torchgeo#1648]
### Description

I spent time trying to reproduce the `EuroSAT.plot()` when inferencing a single sample, and couldn't get the `sample["label"]` and the label shown in the plot to agree. To get them to agree it is necessary to perform `sorted(EuroSAT.classes)` which I do not see in the plot implementation (and doesn't make sense). 

Examples below:

<img width="783" alt="image" src="https://github.com/microsoft/torchgeo/assets/11855322/254cce89-18d6-4d42-b7d2-f9160f36f656">

<img width="770" alt="image" src="https://github.com/microsoft/torchgeo/assets/11855322/821bfcc1-63db-4a9f-ad5f-ec669ad2fe26">

<img width="844" alt="image" src="https://github.com/microsoft/torchgeo/assets/11855322/c7591870-0381-41de-9a16-bd854b652ce4">

### Steps to reproduce

```
# note my version without normalisation, although this shouldn't affect the plot
class EuroSATDataModule(NonGeoDataModule):
    """LightningDataModule implementation for the EuroSAT dataset.

    Uses the train/val/test splits from the dataset.

    .. versionadded:: 0.2
    """

    mean = torch.zeros(13)
    std = torch.ones(13)

    def __init__(
        self, batch_size: int = 64, num_workers: int = 0, **kwargs: Any
    ) -> None:
        """Initialize a new EuroSATDataModule instance.

        Args:
            batch_size: Size of each mini-batch.
            num_workers: Number of workers for parallel data loading.
            **kwargs: Additional keyword arguments passed to
                :class:`~torchgeo.datasets.EuroSAT`.
        """
        super().__init__(EuroSAT, batch_size, num_workers, **kwargs)


datamodule = EuroSATDataModule(
    batch_size=batch_size, 
    root="data", 
    num_workers=num_workers, 
    download=True,
)


sample = datamodule.test_dataset[0]
label = cast(int, sample["label"].item())
image = sample['image'].unsqueeze(0).to(device)
pred = task(image)
pred_index = int(torch.argmax(pred))

result_str = f"label: {sorted(EuroSAT.classes)[label]}, prediction: {EuroSAT.classes[pred_index]}" # sorted shouldn't be required!
fig = datamodule.test_dataset.plot(sample, suptitle=result_str)
```

### Version

0.5.0

**Comment by adamjstewart** [comment]
`EuroSAT` subclasses `NonGeoClassificationDataset` which subclasses `ImageFolder`. [ImageFolder](https://github.com/pytorch/vision/blob/main/torchvision/datasets/folder.py) sorts directory names to determine labels. Are the label names wrong? We can sort the class names if necessary.

**Comment by robmarkcole** [comment]
Check other classification datasets: 

- https://github.com/microsoft/torchgeo/blob/main/torchgeo/datasets/resisc45.py -> already sorted
- https://github.com/microsoft/torchgeo/blob/main/torchgeo/datasets/ucmerced.py -> already sorted
- https://github.com/microsoft/torchgeo/blob/main/torchgeo/datasets/patternnet.py -> no classes attribute defined

So only eurosat needs updating

**Comment by adamjstewart** [comment]
The fact that instantiating the dataset overwrites `self.classes` makes me kinda not want to set it at all for these datasets. We should also ensure that the order documented in the docstring matches. Thoughts?

**Comment by robmarkcole** [comment]
Agreed it doesn't need to be included, I assumed it was a convenience if you wanted to use them elsewhere. You could for example have 'pretty names' like `Industrial buildings` in the docstring _in order_, and then people could just copy from there if they need them
